### PR TITLE
Site Editor: Refactor the site editor initialization to avoid editedEntity state

### PR DESCRIPTION
**Pull request by youknowriad**
## What?

**Note** This should be considered as a prototype at the moment. It's not yet clear to me if we should move forward with this refactor.

This is a refactor that deprecates the "editedEntity" state from the site editor store and replaces it with regular selector from core-data (a hook is used to simplify) and props being passed top down. (A context can be considered for the props passed top down)

## Why?

The reasons I considered this refactor are multiple:

 - This state is actually duplicating the url which could be fine if the synchronization was correct, except that the synchronization was incorrect requiring us to call `setPage` in random components.
 - @ellatrix is working on a PR to add "side by side" view to the pages list, in this PR, we need a way to render the "editor frame" but without it being controlled from the URL and the `getEditedPostType`... selectors. We want to provide the post type and post id to use to render such a frame. The current PR allows that by making these postType and postID as props.
 - I'm hoping this could simplify some performance work as well.$

While I'm not sure yet that we should land it as is, I'm confident that something like this is needed because the site editor as it exists today is no longer a page that renders a site editor for a given entity in which case an "editedEntity" would make sense in its store, today the site editor is comparable to wp-admin as it can render any kind of page.

## Testing Instructions

This PR impacts the whole site editor so we should test everything. We have tests for most flows but there's a lot to test here but I also want to get opinions on the PR.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-edit-mode/page-panels/reset-default-template.js [hunk 1]
```diff
@@ -32,9 +25,6 @@ export default function ResetDefaultTemplate( { onClick } ) {
 				onClick={ async () => {
 					entity.edit( { template: '' }, { undoIgnore: true } );
 					onClick();
-					await setPage( {
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-edit-mode/page-panels/reset-default-template.js:35]: This is the kind of weird action calls that were needed because of the broken synchronization.

File: packages/edit-site/src/hooks/use-edited-entity-from-params.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { store as coreDataStore } from '@wordpress/core-data';
+
+const postTypesWithoutParentTemplate = [
+	'wp_template',
+	'wp_template_part',
+	'wp_block',
+	'wp_navigation',
+];
+
+export default function useEditedEntityForParams( params ) {
```
Comment (youknowriad) [inline packages/edit-site/src/hooks/use-edited-entity-from-params.js:14]: This is the hook that can be used to resolve what postType and postId to use to render an editor frame for the given (postType, postId) tuple. If none is provided we fallback to the home page.

File: docs/reference-guides/data/data-core.md [hunk 3]
```diff
@@ -150,6 +150,10 @@ _Returns_
 
 -   `undefined< 'edit' >`: Current user object.
 
+### getDefaultTemplateId
```
Comment (youknowriad) [inline docs/reference-guides/data/data-core.md:153]: This will be extracted to its own PR, it just replaces the adhoc template lookup calls with proper core-data selector/resolver.

**Comment by youknowriad** [comment]
ok, so I've thought about this more during the weekend, and I think I've come to an important realization. 

First, it's not just the "getEditedPostType" and "getEditedPostId" that are specific to a given "frame/editor". There are more things that are not really global to the whole site editor at the moment: all the UI state (inserter, panels...) are also specific to a given frame and not something site editor state.

I've come to realize that there's a better path forward here: I think we need to split the site editor in two parts:
 - One package that is about the navigation, the sidebars... 
 - One other package that represents basically: `<Editor postType={postType} postI={postId} />`
 
Also that second package can actually be used within **edit-post** exactly as is. (maybe some config to say hide global styles...) and we'll end up with the exact same editor in both finally closing any gaps between these two.

**Comment by SaxonF** [comment]
> I've come to realize that there's a better path forward here: I think we need to split the site editor in two parts:

@youknowriad that sounds pretty great to me.

**Comment by ntsekouras** [comment]
>One package that is about the navigation, the sidebars...

In my mind most of the UI is coupled with current frame, except of course the main navigation sidebar. Can you elaborate a bit more about what this package should contain?

**Comment by youknowriad** [comment]
> In my mind most of the UI is coupled with current frame, except of course the main navigation sidebar. Can you elaborate a bit more about what this package should contain?

IMO, and AFAIK, there's no coupling at all right now between the frame and the sidebar. They only communicate through core-data entities. Both depend on the "post type" "post id" from the URL though

**Comment by youknowriad** [comment]
closing this PR as this work has shifted to another approach described in https://github.com/WordPress/gutenberg/issues/52632#issuecomment-1801436612
